vm/src/frame.rs

```diff
@@ -586,6 +588,7 @@ impl Frame {
                 let obj = vm.ctx.new_function(code_obj, scope, defaults);
 
                 vm.ctx.set_attr(&obj, "__annotations__", annotations);
+                vm.ctx.set_attr(&obj, "__doc__", doc);
```

Comment:
Instead of implementing setattr here, we could implement it at the compiler part, by using the instruction StoreAttr on the just created function. This will also prevent empty docstrings in for example lambda functions.